hack/cleanup/cleanup_oapi.py

```diff
@@ -0,0 +1,130 @@
+    parser.add_argument('--days', action='store', default=30, type=int, help='Clean up omi older than X days (default = 30 days)'),
+    parser.add_argument('--owner', action='store', default='123456789123', type=str, help='Outscale account owner ID (fictional default account number)'),
+    parser.add_argument('--imageId', action='store', default='ami-00000000', type=str, help='Outscale Image Id',)
+    parser.add_argument('--endpointUrl', action='store', default='https://fcu.eu-west-2.outscale.com', type=str, help='Outscale Endpoint Url',)
+    parser.add_argument('--imageName', action='store', default='debian123', type=str, help='Outscale Image Name',)
+    return parser.parse_args()
+
+# checkImageIdAttached check imageId exist
+def checkImageIdAttached(gateway, imageId, accountOwnerImageId):
+    """Check Image Id"""
+    vms = gateway.ReadVms().get('Vms', [])
+    vmsImageId = [vm["ImageId"] for vm in vms if 'ReservationId' in vm]
+    for vmImageId in vmsImageId:
+        if vmImageId == imageId:
+            print("Get Vm Image {}".format(imageId))
+            return True
+    return False
+
+# deleteImageId delete imageId
+def deleteImageId(gateway, imageId, accountOwnerImageId):
+    """Delete Image Id """
+    res = gateway.DeleteImage(ImageId=imageId)
+    if "Errors" in res:
+        print("Can not delete Image {} because {}".format(imageId, res))
+    print("Delete ImageId {}".format(imageId))
+
+# getImageId retrieve imageId        
+def getImageId(gateway, imageName, accountOwnerImageId):
+    """Get Image Id"""
+    for image in gateway.ReadImages(Filters={'ImageNames': [imageName], 'AccountIds': [accountOwnerImageId]}).get('Images', []):
+        imageId = image['ImageId']
+        print("Get ImageId {}".format(imageId))
+        return imageId
+
+# checkImageIdExist check imageId exist
+def checkImageIdExist(gateway, imageId, accountOwnerImageId):
+    """Check Image Id exist"""
+    for image in gateway.ReadImages(Filters={'AccountIds': [accountOwnerImageId]}).get('Images', []):
+        if image['ImageId'] == imageId:
+            print("ImageId {} does exist".format(imageId))
+            return True
+    print("ImageId {} does not exist".format(imageId))
+    return False
+    
+# deleteImage delete image.
+def deleteImage(gateway, accountOwnerImageId, daysCount):
+    """Delete Image"""
+    for image in gateway.ReadImages(Filters={'AccountIds': [accountOwnerImageId]}).get('Images', []):
+        imageId = image['ImageId']
+        creationDate = image['CreationDate']
+        print("Get ImageId {} in {}".format(imageId, creationDate))
+        checkImageToBeDelete(gateway, accountOwnerImageId, creationDate, imageId, daysCount)
+
+# deleteUnattachedImage delete unattached image
+def deleteUnattachedImage(gateway, imageId, accountOwnerImageId):
+    """Delete Unattahed Image"""
+    imageIdAttached = checkImageIdAttached(gateway, imageId, accountOwnerImageId)
```

Comment:
You can only check image attached to the account from which you making the call, it does not meean that nobody uses it

Reply:
Ok done